crates/node/src/migration_service/web/server.rs

```diff
@@ -0,0 +1,240 @@
+use std::{convert::Infallible, sync::Arc};
+
+use hyper::{service::service_fn, Body, Response, StatusCode};
+use tokio::{
+    net::{TcpListener, TcpStream},
+    sync::watch,
+};
+use tokio_rustls::TlsAcceptor;
+use tokio_util::sync::CancellationToken;
+
+use crate::{
+    config::WebUIConfig,
+    keyshare::Keyshare,
+    migration_service::{types::MigrationInfo, web::authentication::authenticate_peer},
+};
+
+use super::types::{ExpectedPeerInfo, WebServerState};
+
+pub async fn start_web_server(
+    web_server_state: Arc<WebServerState>,
+    config: WebUIConfig,
+    migration_state_receiver: watch::Receiver<MigrationInfo>,
+    p2p_private_key: &ed25519_dalek::SigningKey,
+    cancellation_token: CancellationToken,
+) -> anyhow::Result<()> {
+    let (server_config, _client_config) = mpc_tls::tls::configure_tls(p2p_private_key)?;
+
+    tracing::info!(
+        host = %config.host,
+        port = %config.port,
+        "Attempting to bind web server to host",
```

Comment:
Apparently, we should start log messages with a lowercase letter:
```suggestion
        "attempting to bind web server to host",
```